src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,50 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    public static final String lineSeparator = System.lineSeparator();
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] supplyTotal = {0};
+        int[] buyTotal = {0};
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                processLine(line, supplyTotal, buyTotal);
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error reading data from the file " + fromFileName, e);
+        }
+        int result = calculateResult(supplyTotal[0], buyTotal[0]);
+        writeReport(toFileName, supplyTotal[0], buyTotal[0], result);
```

Comment:
```suggestion
        List<String> fileData = readFile(fromFileName);
        String report = makeReport(fileData);
        writeToFile(toFileName, report);
```
Make this method like this